contracts/pToken.sol

```diff
@@ -75,7 +81,19 @@ contract PToken is
             recipient != address(this) ,
             "Recipient cannot be the token contract address!"
         );
+        uint256 gasReserve = 1000; // enough gas to ensure we eventually emit, and return
         _mint(recipient, value, userData, operatorData);
+        if (userData.length > 0) {
+            // pNetwork aims to deliver cross chain messages successfully regardless of what the user may do with them.
+            // We do not want this mint transaction reverting if their receiveUserData function reverts,
+            // and thus we swallow any such errors, emitting a `ReceiveUserDataFailed` event instead.
+            // This way, a user also has the option include userData even when minting to an externally owned account.
+            // Here excessivelySafeCall executes a low-level call which does not revert the caller transaction if the callee reverts,
+            // with the increased protection for returnbombing, i.e. the returndata copy is limited to 256 bytes.
+            bytes memory data = abi.encodeWithSelector(IPReceiver.receiveUserData.selector, userData);
+            (bool success,) = recipient.excessivelySafeCall(gasleft() - gasReserve, 0, 256, data);
```

Comment:
A further improvement would be to reduce 256 since we do not use the return value. We may also consider the possibility to tweak excessivelySafeCall removing the parameter, and the copy of return data... maybe this last option is too aggressive though

Reply:
Agreed after our discussion in private.